vendor/engines/badges_engine/app/helpers/badges_engine/base_helper.rb

```diff
@@ -0,0 +1,65 @@
+module BadgesEngine
+  module BaseHelper
+    def show_badges
+      if @badges.empty?
+        render partial: 'badges_engine/shared/default_badge_message'
+      else
+        render partial: 'badges_engine/shared/badge', collection: @badges
+      end
+    end
+
+    def unknown_image?(badge)
+      'badge-unknown' unless current_user.badges.include?(badge)
+    end
+
+    def badge_earn_icon(badge)
+      if params[:controller] == 'badges' and badge.users.include? current_user
+        content_tag :div, nil, class: 'badge-earned-icon'
+      end
+    end
+
+    def badge_totals(user)
+      "#{ user.badges.size }/#{ @badges.size }"
+    end
+
+    def render_partial?(users, partial)
+      render partial if users.include?(current_user)
+    end
+
+    def define_badge_column
```

Comment:
mm not sure about the reasons, but this could be completely done with css.... with minimal changes... remember, even you could do something it doesn't mean you must do it... if you think about this, you're adding logic which make harder to read the template ... with almost no value ... we could talk about this later